### PR TITLE
add fetch test from http

### PR DESCRIPTION
**Pull request by dylanhitt**
Partially implements #109. @SimonBaeumer reading from stdin could come from another PR or I could bundle it in to this one. I think after stdin I may shoot over some prototypes for some possible refactors of `test_command.go`. It feels a little strange passing `testPath`, `ctx.Filters`, and `fileName` between many different functions - this could also be addressed in #141 as well.

I'll add some test by the end of the week.

## Checklist

 - [x] Added unit / integration tests for windows, macOS and Linux? 
 - [x] Added a changelog entry in [CHANGELOG.md](https://github.com/commander-cli/commander/blob/master/CHANGELOG.md)?
 - [x] Updated the documentation ([README.md](https://github.com/commander-cli/commander/blob/master/README.md), [docs](https://github.com/commander-cli/commander/blob/master/docs))?
 - [x] Does your change work on `Linux`, `Windows` and `macOS`?

### CODE REVIEW

File: pkg/app/test_command.go [hunk 1]
```diff
@@ -36,6 +38,10 @@ func TestCommand(testPath string, ctx TestCommandContext) error {
 		fmt.Println("Starting test against directory: " + testPath + "...")
```
Comment (SimonBaeumer) [inline pkg/app/test_command.go:38]: What happens if `ctx.Dir` is true and the path is an URL?

**Comment by SimonBaeumer** [comment]
This looks good to me, reading from `stdin` could be handeld inside the switch statement too.
I would like to use a syntax like `echo "..." | commander test -`.

**Comment by SimonBaeumer** [comment]
Your code looks good to me, looking forward for the tests :+1:

File: pkg/app/test_command.go (hunk 1 above)
Comment (dylanhitt) [reply pkg/app/test_command.go:38]: `files, err := ioutil.ReadDir(directory)` will respond with 
```
commander (fetch-test-suites-over-http) $ ./commander test --dir https://raw.githubusercontent.com/commander-cli/commander/master/examples/commander.yaml
Starting test against directory: https://raw.githubusercontent.com/commander-cli/commander/master/examples/commander.yaml...

open https://raw.githubusercontent.com/commander-cli/commander/master/examples/commander.yaml: no such file or directory
```
and 

```
./commander test --dir examples/commander.yaml Starting test against directory: examples/commander.yaml...

Hello
fdopendir: not a directory
```
I think a better error here would be 
```
commander (fetch-test-suites-over-http) $ ./commander test --dir example/ 
Starting test against directory: example/...

Error: Input is not a directory
commander (fetch-test-suites-over-http) $ ./commander test --dir https://raw.githubusercontent.com/commander-cli/commander/master/examples/commander.yaml
Starting test against directory: https://raw.githubusercontent.com/commander-cli/commander/master/examples/commander.yaml...

Error: Input is not a directory
commander (fetch-test-suites-over-http) $ ./commander test --dir examples/commander.yaml Starting test against directory: examples/commander.yaml...

Error: Input is not a directory
commander (fetch-test-suites-over-http) $ ./commander test --dir -
Starting test against directory: -...

Error: Input is not a directory
```

Comment (SimonBaeumer) [reply pkg/app/test_command.go:38]: That looks good, thank you!

**Comment by dylanhitt** [comment]
Hey @SimonBaeumer, I finished adding tests. I'm currently targeting a file that's already in master so I can demonstrate the integration test. I plan to switch over to `remote_http.yaml` once that file is in master. 

One important note is that I did add a testing dependency - [gock](https://github.com/h2non/gock), in order to mock the http request. I'll add url support to docs tomorrow. I would like to have this merged, and the implement stdin in a separate PR, mainly just to switch the integration test file over to `remote_http.yaml`. If you have another suggestion let me know.

Best

**Comment by SimonBaeumer** [comment]
Your pull request looks great! 
Just run `go mod vendor` to add the `gock` dependecies in git :)

Let me know when you are finished, would love to merge this :)

**Comment by dylanhitt** [comment]
Done!

**Comment by SimonBaeumer** [comment]
merge :)

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**